### DISCUSSION TITLE
[!TextBlock.TextProperty] syntax?

### DISCUSSION THREAD

**Discussion by musmuris** [AvaloniaUI/Avalonia#15883] [General]
Just going through the Avalonia UI To Do App tutorial and following down some side links cam to this page:https://docs.avaloniaui.net/docs/concepts/templates/creating-data-templates-in-code 

In that there's some C# code `[!TextBlock.TextProperty]`  that I'm not sure what that syntax is?  Wondered if it was some recent C# syntax I'd not seen/remembered. But hovering over it in VS shows the `!` is an IndexerDescripter operator. Googling that didn't show anything obvious that explains the concept, just the reference pages.  Though as this is a side-quest I didn't look too hard just yet and thought I'd ask if someone can point me at a doc page maybe?

**Comment by rabbitism** [comment]
This is a override of indexer operator and negate operator.

**Reply by rabbitism** [reply]
Docs here https://docs.avaloniaui.net/docs/guides/data-binding/binding-from-code#setting-a-binding-in-an-object-initializer

**Reply by musmuris** [reply]
Ah right AvaloniaObject has the indexer which is how it ties together. 
https://github.com/AvaloniaUI/Avalonia/blob/9be975b0e4874bc5d1cad4872615db40959a2430/src/Avalonia.Base/AvaloniaObject.cs#L101

Thanks!
